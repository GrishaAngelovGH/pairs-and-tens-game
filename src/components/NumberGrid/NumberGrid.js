const NumberGrid = () => {
  const n = 10
  const grid = new Array(n).fill().map(() => new Array(n).fill().map((v, i) => i))

  return (
    <div className="row justify-content-center">
      <div className="col-md-6 border border-3 border-primary rounded shadow">
        {
          grid.map((row, i) => (
            <div key={i} className="row">
              {
                row.map((v, j) => (
                  <div
                    key={`${i}${j}`}
                    role="button"
                    className="col border border-primary text-center fw-bold bg-white"
                  >
                    {v}
                  </div>
                ))
              }
            </div>
          ))
        }
      </div>
    </div>
  )
}

export default NumberGrid