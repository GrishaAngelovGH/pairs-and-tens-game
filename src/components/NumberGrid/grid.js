const hasAvailablePlaceInGrid = grid => grid.flat().map(v => v.value).includes(null)

const generateRandomCoordinates = n => {
  const i = Math.floor(Math.random() * n)
  const j = Math.floor(Math.random() * n)

  return [i, j]
}

const placeNumber = (grid, n, num) => {
  let [i, j] = generateRandomCoordinates(n)

  while (grid[i][j].value !== null) {
    [i, j] = generateRandomCoordinates(n)
  }

  grid[i][j].value = num
}

const generatePair = (grid, n) => {
  const num = Math.floor(Math.random() * n)

  placeNumber(grid, n, num)
  placeNumber(grid, n, num)
}

const generateSum = (grid, n) => {
  placeNumber(grid, n, 2)
  placeNumber(grid, n, 8)
}

const generateGrid = n => {
  const grid = new Array(n).fill().map(() => {
    return new Array(n).fill().map(() => ({ value: null, isClicked: false }))
  })

  while (hasAvailablePlaceInGrid(grid)) {
    const shouldGeneratePair = Math.random() >= 0.5

    shouldGeneratePair ? generatePair(grid, n) : generateSum(grid, n)
  }

  return grid
}

export default generateGrid