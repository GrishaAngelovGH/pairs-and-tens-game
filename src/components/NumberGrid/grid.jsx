const sumPairs = [
  [1, 2, 3, 4],
  [9, 8, 7, 6]
]

const hasAvailablePlaceInGrid = grid => grid.flat().map(v => v.value).includes(null)

const generateRandomCoordinates = n => {
  const i = Math.floor(Math.random() * n)
  const j = Math.floor(Math.random() * n)

  return [i, j]
}

const placeNumber = (grid, n, num) => {
  let [i, j] = generateRandomCoordinates(n)

  while (grid[i][j].value !== null) {
    [i, j] = generateRandomCoordinates(n)
  }

  grid[i][j].value = num
}

const generatePair = (grid, n) => {
  const num = Math.floor(Math.random() * n)

  placeNumber(grid, n, num)
  placeNumber(grid, n, num)
}

const generateSum = (grid, n) => {
  const index = Math.floor(Math.random() * 4)

  placeNumber(grid, n, sumPairs[0][index])
  placeNumber(grid, n, sumPairs[1][index])
}

const generateGrid = n => {
  const grid = new Array(n).fill().map(() => {
    return new Array(n).fill().map(() => ({ value: null, isClicked: false }))
  })

  while (hasAvailablePlaceInGrid(grid)) {
    const shouldGeneratePair = Math.random() >= 0.5

    shouldGeneratePair ? generatePair(grid, n) : generateSum(grid, n)
  }

  return grid
}

export default generateGrid