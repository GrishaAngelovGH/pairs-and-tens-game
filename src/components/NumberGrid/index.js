import NumberGrid from "./NumberGrid"

export default NumberGrid