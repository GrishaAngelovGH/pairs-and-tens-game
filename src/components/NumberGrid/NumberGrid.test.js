import { render } from "@testing-library/react"

import NumberGrid from "./NumberGrid"

test("should render NumberGrid component", () => {
  const view = render(<NumberGrid />)

  expect(view).toMatchSnapshot()
})