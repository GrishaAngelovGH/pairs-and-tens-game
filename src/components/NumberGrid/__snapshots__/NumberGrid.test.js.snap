// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NumberGrid component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row justify-content-center"
      >
        <div
          class="col-md-6 border border-3 border-primary rounded shadow"
        >
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              0
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              1
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              2
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              3
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              4
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              5
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              6
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              7
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              8
            </div>
            <div
              class="col border border-primary text-center fw-bold bg-white"
              role="button"
            >
              9
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row justify-content-center"
    >
      <div
        class="col-md-6 border border-3 border-primary rounded shadow"
      >
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            0
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            1
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            2
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            3
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            4
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            5
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            6
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            7
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            8
          </div>
          <div
            class="col border border-primary text-center fw-bold bg-white"
            role="button"
          >
            9
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
